src/Uno.UI/UI/Xaml/Markup/Reader/XamlObjectBuilder.cs

```diff
@@ -822,6 +821,39 @@ private object BuildLiteralValue(XamlMemberDefinition member, Type propertyType
 
 		private object BuildLiteralValue(Type propertyType, string memberValue)
 		{
+			if (propertyType.GetCustomAttribute<CreateFromStringAttribute>() is { } createFromString)
+			{
+				var sourceType = propertyType;
+				var methodName = createFromString.MethodName;
+				if (createFromString.MethodName.Contains("."))
+				{
+					var splitIndex = createFromString.MethodName.LastIndexOf(".");
+					var typeName = createFromString.MethodName.Substring(0, splitIndex);
+					sourceType = AppDomain.CurrentDomain
```

Comment:
TypeResolver should be able to provide this type of lookup

Reply:
Unfortunately it expects a different format (`ns:Type` or just `Type`), whereas in this case I have full type name

Reply:
This approach did not work due to the change I made in this PR - I added this line - https://github.com/unoplatform/uno/blob/47aed3f70f56edcbced643b9119a7489f3cbfd54/src/Uno.UI/UI/Xaml/Markup/Reader/XamlTypeResolver.cs#L359

This covers the scenario where the xmlns of the file is not standard - e.g. xmlns="using:Uno.UI.Tests.Windows_UI_Xaml_Markup.XamlReaderTests which is something that UWP handles properly. This line prepends the default namespace before the type name so "XYZ" becomes "Uno.UI.......XamlReaderTests.XYZ" - which is needed for Type.GetType(name) to work. However, when the type passed into XamlTypeResolver is already fully qualified, it causes it to become "Uno.UI.......XamlReaderTests.Uno.UI.......XamlReaderTests.XYZ" . Not sure how to resolve this to satisfy both scenarios, as it is not really possible to tell whether the string is a qualified or non-qualified type, other than checking if indexof "." is -1.

However, I added this, which makes a fallback to use the original type and that resolves this:

https://github.com/unoplatform/uno/pull/7535/files#diff-9c43ddb90b5f877f348d20114e11c095881eaea4c50d5df26db6f4d359ee2ae5R378

Reply:
That works as well!